backend/src/libs/packages/api/base-http-api.ts

```diff
@@ -83,17 +83,11 @@ class BaseHttpApi implements HTTPApi {
     return response;
   }
 
-  private async handleError(response: Response): Promise<never> {
-    let parsedException: ServerErrorResponse;
-
-    try {
-      parsedException = (await response.json()) as ServerErrorResponse;
-    } catch {
-      parsedException = {
-        errorType: ServerErrorType.COMMON,
-        message: response.statusText,
-      };
-    }
+  private handleError(response: Response): Promise<never> {
```

Comment:
we also need to catch and rethrow error in service, because we need to override status code, for example if open ai respond with 401, we don't want to respond with 401 because user will be signed out